### DISCUSSION TITLE
LVT datastream attributes

### DISCUSSION THREAD

**Discussion by slinskik** [NASA-LIS/LISF#1088] [Q&A]
Hi LIS team - I was wondering where I can find information on the datastream attributes the LVT expects for non-LIS-generated datasets.  This issue comes up when I try to process a new dataset in LVT.  For example, I was recently trying to process IMERG-Final data using the datastream attributes:  “Rainf_f    1 1  kg/m2s  DN  1 1”, but had to change it to:  “TotalPrecip       1  1 kg/m2   DN  3 1” for LVT to run.  Many thanks for your help with this.

Kim

**Comment by dmocko** [comment]
HI @slinskik 

Try running this command in this directory to see the LVT variables and available units for each datastream:

`LISF/lvt/datastreams> grep -i -A 2 lvt_logsingle */* | more`

e.g.:

```
IMERG/readIMERGdata.F90:   call LVT_logSingleDataStreamVar(LVT_MOC_totalprecip,source,prcp_final,&
IMERG/readIMERGdata.F90-        vlevel=1,units='kg/m2s')
IMERG/readIMERGdata.F90-   ! Now convert from kg/m2s to kg/m2
--
IMERG/readIMERGdata.F90:   call LVT_logSingleDataStreamVar(LVT_MOC_totalprecip,source,prcp_final,&
IMERG/readIMERGdata.F90-        vlevel=1,units='kg/m2') 
IMERG/readIMERGdata.F90-end subroutine readIMERGdata
```

**Reply by slinskik** [reply]
Thanks for the information @dmocko !  

I'm still a little confused.  E.g., for ERA5 air temperature data,  readERA5obs.F90 indicates:

call LVT_logSingleDataStreamVar(LVT_MOC_TAIRFORC,source, tair,&
       vlevel=1,units="K")

From this, I thought the variable name would be "TAIRFORC", but the variable name needed for the datastream attributes table is actually "Tair_f".    I was wondering what I'm missing. 

Many thanks for your help!
